VideosController.js

```diff
@@ -0,0 +1,44 @@
+
+var express = require('express');
+var router = express.Router();
+var Events = require('./Events');
+
+router.get('/', function (req, res) {
```

Comment:
Not familiar with how JS works -- does this assume a persistent connection that waits for events, or is it a more conventional HTTP connection that bundles all the events at once?

Reply:
It's a node http server process that listens on a port

Reply:
Right, but I'm not really following how it's supposed to work -- there is a loop in this request handler that iterates over events, so does that mean all the events get submitted at once and the handler exits, or does the handler wait for new events to come in on the connection? The reason I ask is because that affects the semantics around storing state in the `video` object, especially if the HTTP connection is broken mid-stream.

Reply:
No loop here.  It's just executing a query, looping over the results, and writing a response back.

Reply:
`Events.find()` will search the db and aggregate the events. Once the results are found, it looks like @ericxtang is mapping the array of stored event objects into an object with the following shape:

```js
type VideoIndex = string // concatenation of `manifestId` and `createTime`
type Video = {
  manifestId: string,
  createTime?: Date,
  endTime?: Date,
  success: boolean
}
type VideosResponseBody = {
  [idx: VideoIndex]: Video
}
```
That object result is then returned to the requestor in the response body. Is that what you're asking @j0sh ?

Reply:
Yep -- my misunderstanding was that I thought the events got submitted as part of the HTTP request; I didn't realize that the handler was actually performing a query.